src/sortStudents.ts

```diff
@@ -1,16 +1,75 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'desc' | 'asc';
 
+function findAverageGrade(studentGrades: number[]): number {
+  const sumOfGrades = studentGrades.reduce((sum: number, grade: number) => {
+    return sum + grade;
+  }, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sumOfGrades / studentGrades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  let copyStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyStudents = copyStudents.sort((first, second) => {
+        if (order === 'asc') {
+          return first[sortBy].localeCompare(second[sortBy]);
+        }
+
+        return second[sortBy].localeCompare(first[sortBy]);
+      });
+
+      break;
+
```

Comment:
sort is a mutable method, so let's use const instead of let and return the result immediately
```suggestion
      return copyStudents.sort((first, second) => {
        if (order === 'asc') {
          return first[sortBy].localeCompare(second[sortBy]);
        }

        return second[sortBy].localeCompare(first[sortBy]);
      });

```